src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -122,7 +123,7 @@ private void ProcessReduce(IndexDefinition definition, ObjectInstance definition
         }
 
         private void ProcessMaps(ObjectInstance definitions, JintPreventResolvingTasksReferenceResolver resolver, List<string> mapList,
-            List<HashSet<CollectionName>> mapReferencedCollections, out Dictionary<string, List<JavaScriptMapOperation>> collectionFunctions)
+            List<(HashSet<CollectionName> ReferencedCollections, bool HasCompareExchangeReferences)> mapReferencedCollections, out Dictionary<string, List<JavaScriptMapOperation>> collectionFunctions)
```

Comment:
Consider to create a dedicated type for tuple `(HashSet<CollectionName> ReferencedCollections, bool HasCompareExchangeReferences)` - it's very long and hard to read it

Reply:
Done